src/Birkhoff_polytope.jl

```diff
@@ -0,0 +1,453 @@
+# Get the list of lower bounds.
+function Boscia.get_lower_bound_list(blmo::BirkhoffBLMO)
+    return collect(1:length(blmo.lower_bounds))
+end
+
+# Get the list of upper bounds.
+function Boscia.get_upper_bound_list(blmo::BirkhoffBLMO)
+    return collect(1:length(blmo.upper_bounds))
+end
+
+# Read bound value for c_idx.
+function Boscia.get_bound(blmo::BirkhoffBLMO, c_idx, sense::Symbol)
+    if sense == :lessthan
+        return blmo.upper_bounds[c_idx]
+    elseif sense == :greaterthan
+        return blmo.lower_bounds[c_idx]
+    else
+        error("Allowed value for sense are :lessthan and :greaterthan!")
+    end
+end
+
+## Changing the bounds constraints.
+
+# Change the value of the bound c_idx.
+function Boscia.set_bound!(blmo::BirkhoffBLMO, c_idx, value, sense::Symbol)
+    if sense == :greaterthan
+        blmo.lower_bounds[c_idx] = value
+    elseif sense == :lessthan
+        blmo.upper_bounds[c_idx] = value
+    else
+        error("Allowed values for sense are :lessthan and :greaterthan.")
+    end
+end
+
+# Delete bounds.
+function Boscia.delete_bounds!(blmo::BirkhoffBLMO, cons_delete)
+    for (d_idx, sense) in cons_delete
+        if sense == :greaterthan
+            blmo.lower_bounds[d_idx] = -Inf
+        else
+            blmo.upper_bounds[d_idx] = Inf
+        end
+    end
+end
+
+# Add bound constraint.
+function Boscia.add_bound_constraint!(blmo::BirkhoffBLMO, key, value, sense::Symbol)
+    idx = findfirst(x -> x == key, blmo.int_vars)
+    if sense == :greaterthan
+        blmo.lower_bounds[idx] = value
+    elseif sense == :lessthan
+        blmo.upper_bounds[idx] = value
+    else
+        error("Allowed value of sense are :lessthan and :greaterthan!")
+    end
+end
+
+## Checks
+
+# Check if the subject of the bound c_idx is an integer variable (recorded in int_vars).
+function Boscia.is_constraint_on_int_var(blmo::BirkhoffBLMO, c_idx, int_vars)
+    return blmo.int_vars[c_idx] in int_vars
+end
+
+# To check if there is bound for the variable in the global or node bounds.
+function Boscia.is_bound_in(blmo::BirkhoffBLMO, c_idx, bounds)
+    return haskey(bounds, blmo.int_vars[c_idx])
+end
+
+# Has variable an integer constraint?
+function Boscia.has_integer_constraint(blmo::BirkhoffBLMO, idx)
+    return idx in blmo.int_vars
+end
+
+## Optional
+
+function Boscia.check_feasibility(blmo::BirkhoffBLMO)
+    for (lb, ub) in zip(blmo.lower_bounds, blmo.upper_bounds)
+        if ub < lb
+            return Boscia.INFEASIBLE
+        end
+    end
+    # For double stochastic matrices, each row and column must sum to 1
+    # We check if the bounds allow for feasible assignments
+    n0 = blmo.dim
+    n = n0^2
+    # Initialize row and column bound tracking
+    row_min_sum = zeros(n)  # minimum possible sum for each row
+    row_max_sum = zeros(n)  # maximum possible sum for each row
+    col_min_sum = zeros(n)  # minimum possible sum for each column
+    col_max_sum = zeros(n)  # maximum possible sum for each column
+
```

Comment:
I think the max sums should be equal to the all one. Think about the case there we have no integer variables in a given row and column.